core/metadata_handler.cpp

```diff
@@ -0,0 +1,127 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+/*
+ * Copyright (C) 2020, Raspberry Pi (Trading) Ltd.
+ *
+ * metadata_hander.cpp - class for handling metadata
+ */
+
+#include "metadata_handler.hpp"
+
+MetadataHandler::MetadataHandler(VideoOptions const *options)
+	: options_(options), still_options_(nullptr), buf_metadata_(std::cout.rdbuf()), of_metadata_(), fmt_(),
+	  first_write_(true), video_mode_(true)
+{
+	if (options != nullptr)
+		fmt_ = options_->metadata_format;
+}
```

Comment:
I can see how this has all come out of the code that we currently have, but at the same time it does leave some things looking a little awkward. For example, the uncomfortable co-existence of video and still modes.

Thinking out loud a bit, I wonder about removing the `options_` field and replacing it with `video_options_`. Then we have two constructors, one as above that takes a `VideoOptions` (and sets the `video_options_` field) and another that takes a `StillOptions` (and sets the `still_options_` field). We could probably also lose the `video_mode_` flag, as only one of the options is ever set. Would that be a bit neater?